frontend/src/tests/instructor-permission-test.js

```diff
@@ -53,21 +124,80 @@ export function instructorsPermissionTests(api) {
         });
     });
 
-    it.todo("can't update instructors except for self (i.e. active user)");
+    it("can't update instructors except for self (i.e. active user)", async () => {
+        let respFetchSelf = await apiGET(`/instructor/active_user`);
+        expect(respFetchSelf).toHaveStatus("success");
+        const self = respFetchSelf.payload;
+
+        let respFetchInst = await apiGET(`/instructor/instructors`);
+        expect(respFetchInst).toHaveStatus("success");
+        let otherInst = null;
+        respFetchInst.payload.forEach((inst) => {
+            if (self.id !== inst.id) {
+                otherInst = inst.id;
+            }
+        });
+
+        if (otherInst === null) {
+            throw Error("There is no other instructor other than self!");
+        }
+
+        let modifiedOtherInst = { ...otherInst };
```

Comment:
I am a bit confused here. Isn't `otherInst` just `inst.id`, which is an ID value?

Reply:
You are right. This is a mistake. otherInst should have been assigned to the instructor user object.